vavr/src/main/java/io/vavr/control/Try.java

```diff
@@ -631,7 +638,45 @@ default boolean isSingleValued() {
     }
 
     /**
-     * Consumes the value if this is a Success.
+     * Consumes the cause if this is a {@link Try.Failure} and the cause is instance of {@code X}.
+     *
+     * <pre>{@code
+     * // (does not print anything)
+     * Try.success(1).onFailure(Error.class, System.out::println);
+     *
+     * // prints "Error"
+     * Try.failure(new Error())
+     *    .onFailure(RuntimeException.class, x -> System.out.println("Runtime exception"))
+     *    .onFailure(Error.class, x -> System.out.println("Error"));
+     * }</pre>
+     * 
+     * @param exceptionType the exception type that is handled
+     * @param action an excpetion consumer
+     * @param <X> the exception type that should be handled
+     * @return this
+     * @throws NullPointerException if {@code exceptionType} or {@code action} is null
+     */
+    @GwtIncompatible
```

Comment:
@ruslansennov this time I did not forget it 🎉

Reply:
:smile: it doesn't matter, because `vavr-gwt` doesn't working now (caused by changes in `FutureImpl`). Seems we should keep both branches: `master` and, say, `v.0.9.x`. Every commit in `master` which is belongs to minor release should be cherry-picked immediately for obtain actual minor snapshot version.